stl/inc/__msvc_int128.hpp

```diff
@@ -1367,6 +1375,21 @@ struct _Signed128 : _Base128 {
         _Word[1] |= _That._Word[1];
         return *this;
     }
+
+    _NODISCARD constexpr explicit operator long double() const noexcept {
+        bool _Negative = false;
+        _Signed128 _Copy{*this};
+        _Copy._Strip_negative(_Negative);
+        long double _Result = static_cast<long double>(_Copy._Word[0]) + static_cast<long double>(_Copy._Word[1]) * 18446744073709551616.0;
+        if (_Negative) {
+            _Result = -_Result;
+        }
+        return _Result;
+    }
+    template <class _Ty, enable_if_t<is_floating_point_v<_Ty>, int> = 0>
+    _NODISCARD constexpr explicit operator _Ty() const noexcept {
+        return static_cast<_Ty>(static_cast<long double>(*this));
+    }

```

Comment:
I'm not sure whether this correctly covers the minimal value of `_Signed128` for which unary `operator-` triggers UB.
```suggestion
    template <class _Ty, enable_if_t<is_floating_point_v<_Ty>, int> = 0>
    _NODISCARD constexpr explicit operator _Ty() const noexcept {
        if (static_cast<int64_t>(_Word[1]) < 0) {
            return -static_cast<_Ty>(-static_cast<_Unsigned128>(*this));
        } else {
            return static_cast<_Ty>(_Word[1]) * static_cast<_Ty>(18446744073709551616.0) + static_cast<_Ty>(_Word[0]);
        }
    }
```

Reply:
I think we can probably start by converting `_Signed128` to a `bool` and a `_Unsigned128` like https://github.com/ELC-lang/ELC/blob/0f971b0686b0fb8c93c526f1ea1606e227c1471b/parts/header_file/files/elc/_files/base_defs/base_defs/math.hpp#L175

Reply:
how do ya think abt [`22a0383` (#3559)](https://github.com/microsoft/STL/pull/3559/commits/22a03832097ae65866f74055f841be5162dccdde)?